ckanext/recombinant/views.py

```diff
@@ -262,7 +275,22 @@ def data_dictionary(dataset_type):
     blob = StringIO()
     book.save(blob)
     response = Response(blob.getvalue())
-    response.headers['Content-Type'] = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+    # (canada fork only): modify response headers for Microsoft Edge
+    content_type = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+    disposition_type = 'inline'
+    user_agent_legacy = request.get('headers', {}).get('User-Agent')
+    user_agent = request.get('headers', {}).get('Sec-CH-UA', user_agent_legacy)
+    if user_agent and (
+        "Microsoft Edge" in user_agent or
+        "Edg/" in user_agent or
+        "EdgA/" in user_agent
+        ):
+            content_type = 'application/octet-stream'
+            disposition_type = 'attachment'
+    response.headers['Content-Type'] = content_type
+    response.headers['Content-Disposition'] = '{}; filename="{}.xlxs"'.format(
+        disposition_type,
+        dataset_type)
```

Comment:
It's worth putting this into a function to avoid the duplication

Reply:
Moved to method `_xlsx_response_headers`, returning tuple of `content_type` and `disposition_type`